merkle/src/tree_impl.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use super::*;
+use std::marker::PhantomData;
+
+const DEGREE: usize = 2;
+
+/// Special complete binary merkle tree that is compatible with
```

Comment:
Is compatibility important?

Reply:
Compatibility is more referring to the fact that a mina merkle tree puts all leaf nodes at the same height/depth, while in [other implementations](https://crates.io/crates/merkle-cbt), it's not required, thus the tree can be more compact. And because mina uses a depth-aware hash algorithm, I think the latter will break compatibility.